cibuildwheel/macos.py

```diff
@@ -82,6 +98,8 @@ def call(args, env=None, cwd=None, shell=False):
             installation_bin_path,
             env['PATH'],
         ])
+        if config.identifier.startswith('pp') and 'MACOSX_DEPLOYMENT_TARGET' not in env:
+            env['MACOSX_DEPLOYMENT_TARGET'] = '10.13'
```

Comment:
This setting is to late. If you build `pypy` before set `MACOSX_DEPLOYMENT_TARGET` then it is build against current version of system, so you cannot use it to build wheels for lower version of system.

Reply:
@Czaki But I/we don't build PyPy. I just downloaded the pypy.org binaries. Just like we don't build CPython, right?
So I don't really understand how this is too late?

Reply:
OK, no, I found it: https://github.com/python/cpython/blob/62dfd7d6fe11bfa0cd1d7376382c8e7b1275e38c/Lib/distutils/spawn.py#L94-L113

So, since `distutils` (wrongly) believes PyPy only targets macOS version >= 10.14, it errors because the manual override of `MACOSX_DEPLOYMENT_TARGET` is too low :-|

Thanks again!

Reply:
@Czaki No, something's still wrong. Can you explain why this doesn't work?
I thought setting `MACOSX_DEPLOYMENT_TARGET` higher than the internal constant inside of the Python (or PyPy) interpreter should work!?

But I get the wrong filenames: https://circleci.com/gh/joerick/cibuildwheel/2324
Any idea why?

Reply:
you overwrite the macosx deployment target (it is safe?) in sysconfig_pypy.py
But later you set `MACOSX_DEPLOYMENT_TARGET`so it reads target from environment variable and set name to 10.13

Reply:
> you overwrite the macosx deployment target (it is safe?) in sysconfig_pypy.py

Yes, it's safe. I asked on the PyPy IRC :-) They've already changed it for the next release.

> But later you set `MACOSX_DEPLOYMENT_TARGET`so it reads target from environment variable and set name to 10.13

Yes indeed. I might change that later, but so I was expecting file names like `'{package_name}-{package_version}-pp272-pypy_41-macosx_10_13_x86_64.whl'`. But the tests fail because the file name `'spam-0.1.0-pp272-pypy_41-macosx_10_7_x86_64.whl'` is produced. (See https://circleci.com/gh/joerick/cibuildwheel/2324)
Or am I going crazy? :)

Reply:
Ok, it looks like I misunderstand log. I try to check this from the macbook laptop (It will take time)

Reply:
THANK YOU! :-)

Sorry if you never wanted that, but it seems you've become `cibuildwheel` expert on macOS versioning ;-)

Reply:
OK. I might have found it. It's because your PR has not been released yet?

Cfr. https://github.com/pypa/wheel/blob/0.33.6/wheel/pep425tags.py#L108 vs. https://github.com/pypa/wheel/blob/a7510b6510ab3f1768de7eb75d7a6d06303bee24/wheel/pep425tags.py#L181?

Reply:
Yes. I also realized it, a moment ago. Also pypa/wheel#317 need to be merged.

Reply:
Right, thanks a lot for the help!

I guess the bugfix is a detail, in this context :-)